core/agentbinary/types.go

```diff
@@ -6,11 +6,10 @@ package agentbinary
 import (
 	"fmt"
 
-	"github.com/juju/version/v2"
-
 	"github.com/juju/juju/core/arch"
 	coreerrors "github.com/juju/juju/core/errors"
 	"github.com/juju/juju/internal/errors"
+	"github.com/juju/juju/internal/version"
```

Comment:
Beyond `internal/errors` which is borderline. Having internal packages in core doesn't pass the smell test for me.

Reply:
Can move to `core/version` which exists already, and was importing `juju/version/v2` already